discovery/gossiper.go

```diff
@@ -991,8 +1154,74 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []l
 			if routing.IsError(err, routing.ErrOutdated,
 				routing.ErrIgnored) {
 
+				// The edge will get rejected if we already
```

Comment:
Nice! 

However, the indentation is a bit much. Would recommend rolling this into it's own function. This isn't a blocker on the PR though.

Reply:
Added TODO